Reworks Syndicate Supply Depot

Kyep opened the pull request:
This is a rework of the Syndicate Supply Depot, aimed at fixing its problems:
1) Current depot is too predictable. It is exactly the same every round, and its invulnerable walls mean it can only be entered via one route. Overall, the experience is the same every time. Little replay/exploration value.
2) Since it is so predictable, it is also easy to cheese. You can simply drop a maxcap bomb or two on it, kill everything, proceed to loot the entire thing, and walk out. 
3) Even though it can be easily cheesed, it never reacts to your actions, no matter what you do. It is entirely passive.

_This PR reworks the depot into something more fun, challenging, and interesting. In particular:_
1) New depot is unpredictable. The defenses, and loot, are randomized every round. Even the connections between rooms change. Some of the differences are big enough to change your approach by themselves. 
2) New depot is MUCH harder to cheese. Aside from the big random factor, several countermeasures have been taken specifically to prevent it from being easily cheesed.
3) New depot actually reacts to your actions. The more of the defenses you destroy, the more the Syndicate gets worried and the more determined they get to kill you. This means it is a bad idea to attempt to simply kill everything. Instead, you have to pick and choose which rooms you want to try and clear, weighing cost versus benefit.
4) New depot recognizes, and has unique interactions for, syndicate agents. 
 
Screenshot (shows one possible layout - there are many!):
![depot_v_jun_27](https://user-images.githubusercontent.com/16434066/41993310-e6ecc92c-79ff-11e8-8925-7bf978d79664.PNG)



🆑 Kyep
tweak: Replaced the Syndicate Supply Depot with a new version, rebuilt from scratch, that is much more challenging/interesting.
fix: It is no longer possible to walk straight through decoy AI cores as if they were not there.
fix: Syndicate grenade launcher turrets now actually launch grenades. 
/🆑

Kyep commented:
@Spartan6 Currently:
- The depot has an 'alert level'.
- It starts at zero, and increases by one each time you take an action that draws the attention of the Syndicate, such as killing one of the syndie officers, or certain other scripted actions.
- At level 1 and higher, the zone is under a constant fire alarm effect, with the flashing red lights. A very dangerous syndicate patrol bot, equipped with a grenade launcher, also spawns and starts moving around the outside of the depot, looking for intruders.
- At level 2 and higher, the bot is rerouted such that it may path directly through the depot (engaging you en-route), doors are opened/closed to change up the level pathing, and a squad of nukeop-level syndicate NPCs teleports in around the asteroid so that if you try to escape, they will intercept you.
- If you make it to level 3, the syndicate declare the depot as lost to hostile forces, and arm the self-destruct. While not quite as dangerous as a nuke, it will more or less physically destroy the entire depot in one explosion, several times the size of a maxcap toxins bomb.

KingPhilipIII commented:
Looks neat. Define “unique” interactions for agents though. Nonlethal or what?

Kyep commented:
I don't want to spoil all the possible interactions, but here are a few examples:
- Having an agent ID allows you to open doors, and to use the door control computer.
- Being a traitor (the role) gives you special results when using the syndicate communications computer.
- Having an extremely high level syndicate ID card lets you take control of depot infrastructure, such as using the turret control system to turn off the turrets.

KingPhilipIII commented:
Does that mean I could abandon the station and become a Syndicate Depot Officer instead? Cool. And I’d like to figure it all out for myself, my main concern was if I’d be shot on sight by the turrets like they used to do. 

Also I can’t tell because it’s green. Is the space pod in the Syndicate paint scheme or no?

Kyep commented:
The mobs, turrets etc are hostile to everyone, including syndi agents. You could fight your way through the depot and claim it as yours. But it would be a struggle. I decided against giving syndie agents free run of the depot, on the grounds that it would allow them to pick up too much loot too easily.

Yes, the pod is a syndi pod (a new type of pod added in my wild west PR). It is like a normal pod except it uses syndi colors. The pod has a 50% chance to spawn. If it does spawn, it has a 50% chance to be unlocked/usable.

KingPhilipIII commented:
Damn. I can see why you did but still a letdown

Kyep commented:
To be fair, syndie agents have several advantages over normal crew in taking on the depot.
- They may (depending on RNG) be able to emag the teleporter, and lock on to the syndie depot beacon, teleporting straight there. This bypasses the need for a hardsuit. If they do choose to use a hardsuit, syndie hardsuits are armored, which helps.
- They have access to ballistic weapons, which are very good against the depot NPCs.
- They can get agent ID cards, which let them through almost all depot doors.
- They can get thermal glasses, which let them see the locations of defenses in advance, and thus make better decisions about which rooms they want to engage and which they want to bypass.
- They are more likely to find the gear useful in general.

I could go on, but you see my point? Syndie gear works pretty well in the depot. Yeah, it won't be a cakewalk, even for a syndie agent, but they do have some advantages there.

KingPhilipIII commented:
I understand completely, my disappointment wasn’t ‘No free loot!’ It was “No prancing around the depot feeling like a badass”

And to be fair some of those are still paltry at best because we face an issue in just how absolutely punishing ballistic weapons are, and as they should be, but one fuckup during an engagement and unless you’re a tatorling that’s it, done for.

Kyep commented:
The depot is meant to be brutal. It is meant to stand up to the cheesiest of cheese and the most powerful of powergaming. It is meant to handle whatever you can throw at it and still provide a decent challenge. As such, it is quite unforgiving. It is very easy to die near-instantly in the depot. Recognize it for the metal deathtrap that it is ;)

KingPhilipIII commented:
I welcome Darksouls 13 with open arms.

Kyep commented:
I could... but there are already several other counters to girder cheese in the mission. The turrets are the easiest defenses to deal with. They're mostly there as a delaying tactic.

KingPhilipIII commented:
Is it possible to disarm the self destruct if it can be reached in time?

Kyep commented:
The self destruct exists for two reasons: (A) to be a trap, something players can activate at any time by certain very silly actions. (B) to be a final failsafe, designed to stop players from being able to murderbone the entire depot and loot everything, no matter how much force/firepower they bring.

Neither purpose is served by allowing the self destruct to be deactivated once it is engaged. So, currently, it is not possible to deactivate the self destruct. All you can do is run. Quickly.

KingPhilipIII commented:
I honestly see the depot being attacked by the crew and becoming like this massive offensive, with enormous preparation of arming the crew, setting up a teleporter for rapid looting and evacuation, etc.

Kyep commented:
I would actually like to see that.
Here is how I imagine it going:
1. First guy walks in by himself. Falls for a trap in the first room. Dies. The syndicate now know you are here. BLUE alert is declared in the depot.
2. Second guy walks in. In a fully-fit durand. Gets recognized by a syndie as overwhelming threat (mechas = dangerous). This gets the alert level kicked up to RED in the depot.
3. Second and third guy together kill that syndie, then another, then an officer. Then they start looting the depot's armory. They're making good progress, but at this point the depot Syndies realize they're hopelessly outmatched. They order a base-wide lockdown, and activate the self destruct.
4. The raiders can't get out in time, and all of them end up dying in the self-destruct.

I'm not saying it can't be done. I'm just saying: be careful how much force you bring. If you try to cheese it by bringing massive numbers, or overwhelming firepower, the depot's AI will recognize this and kick things up to 11 for you.

Kyep commented:
New Feature: Visitor Mode:
- To activate visitor mode, you must be a real syndie agent (merely having a syndicate tech ID is not enough). Emag the teleporter, lock onto the syndie depot beacon (only present in 50% of shifts), and teleport there. Once you are there, use the syndicate comms computer to the south of the teleport room to request a depot visitor pass. Do not open any locker or loot anything before requesting the pass, or you will be recognized as a con man, your request will be rejected, and the alert level will rise.
- Once you are issued a pass, the rest of the syndie forces in the depot become neutral to you. Traps, however, will still kill you. You're expected to be smart enough to avoid those. Anyone who comes with you requires their own visitor pass. Your pass does not protect them. Again: only real agents can get a pass. Each person must get their own pass. You cannot get a pass for them, nor can you give them your pass.
- While any pass is active, the depot enters visitor mode. The depot AI will declare a Code White (ie: visitors), and all loot containers will be locked, to prevent you from making off with anything. While it is POSSIBLE for you to force your way into a container anyway, and steal stuff, this is extremely risky. Loot containers have sensors, and the depot AI will know if you try it. The Syndicate are far harsher on traitors than NT is. They will immediately cancel ALL visitor passes, declare ALL visitors KIA, hike the depot to code red, remotely reprogram their doors to no longer accept your ID/access, and more. All admins also get a notification that a traitor has robbed the syndie depot, so don't be surprised if every other antag in the round gets an optional objective to kill you.
- While you could in theory get a pass, walk around the depot, try to find the highest concentration of loot boxes, and then try to steal them all before fleeing... in reality surviving this is going to be nigh impossible. Aside from merely getting out of the depot, which will be hard (did I mention: hand teles no longer work in the depot?), you can face retribution even after getting back to the station. Stealing from the depot as a syndicate agent notifies all online admins that a member of the syndicate has betrayed their employers, so don't be surprised if you end up with a whole team of players trying to hunt you down and kill you.

New Feature:  New Loot System
- Almost all existing loot spawners were replaced with a series of consolidated spawners, ranked 1 through 4, with 1 being lowest level, and 4 highest.
- The idea here is that it means greater randomness to loot, whilst also ensuring that the best loot is always in areas with higher risk.
- It also greatly simplifies the amount of random_spawner.dm entries.

KingPhilipIII commented:
And if the crew attacks while traitors are visiting? How does the AI treat them?

Kyep commented:
The AI will attack them.
It will assume that they and the traitors are working together to infiltrate the depot... and respond accordingly.

KingPhilipIII commented:
Snap this is like your terror spider PR in complexity and I love it.

taukausanake commented:
Hand teles don't work in the depot or in the immediate area around the depot? Could you escape the depot and hand tele away or something?

imsxz commented:
>using NPC's to make it harder to cheese

lol.

Kyep commented:
@taukausanake Correct. As of latest commit, hand teleporters do not function in any area with the "tele_proof" flag. This includes the depot, the asteroid around the depot, and the terror spider away mission.

Kyep commented:
Latest Changes:
- Add: All computers are now locked out on red/higher alert.
- Add: Computer lockout also causes all ongoing effects for that computer (e.g: tele beacon) to be deactivated.
- Fix: Depot AI core mob (and decoy ai cores in general) can no longer be pushed.
- Add: Teleport computer now has a secondary function - creating a permanent one-way teleport portal to somewhere on the station zlevel. The only ways to activate this are to put the depot in visitor mode, OR be an admin. The portal is orange, appears next to the tele console, and teleports you to a random location on the station zlevel. The portal is deactivated if peaceful mode ends.
- Tweak: Much refactoring of computer code to more heavily leverage proc inheritance.
- Tweak: Admins now require advanced interaction to be enabled in order to interact with computers as ghosts. 
- Tweak: Refactored "damage computer" code.
- Tweak: Mechs can only increase the alert level by one.

KingPhilipIII commented:
What happens if you kill the depot AI core?

Wait. Lemme guess. Depot goes to delta, you’re locked in the core and kaboom?

Kyep commented:
@KingPhilipIII Nah. Like most hostile actions you can take in the depot, it just raises the alert level by one. There are a few actions which raise it by more than one, or cause the depot to go into self-destruct mode immediately, but they're rare. Typically, those actions involve doing something massively dumb, like betraying the Syndicate after they've granted you a pass as an agent, or shooting the fusion reactor until you crack its protective case, etc.

KingPhilipIII commented:
Are there ways to reduce the alert level or does it gradually decrease over time if further hostilities are not detected, etc etc?

KingPhilipIII commented:
I’m not asking for a comprehensive list, because discovery is half the fun, it’s just curiosity.

Kyep commented:
There is currently no way to reduce the alert level, short of adminbus. The reason for this is that I have to assume that sooner or later, everyone attempting the depot will know all of its mechanics. So, if I put in some sort of secret method to reduce its alert level... sooner or later, everyone will use it.

For example, earlier I was debating having a syndicate comms relay in the depot. Breaking it would prevent the use of the syndie comms console to contact the syndicate. BUT it would also prevent the syndicate from summoning backup from another base at red alert level. The issue I had was that the comms room is connected to an external wall. It is very possible for someone to break into that room first every time, destroy that relay first every time, and then proceed to attack the rest of the depot with one of the three alert level's challenge effectively removed. That's bad. I don't want to be creating puzzles for which the solution is learning what to do by rote, or reading the solution on Github.

A slightly better version I was debating adding was having a telecomms shed on the depot with an officer in it. Killing an officer always raises the alert level by one. So, you'd be making a choice: is the negation of the syndicate backup worth raising the alert level by one for? That could be an interesting gameplay decision. 

Anyway, that's the approach I'm trying to take: I don't want there to be any clear "best" strategy for the depot. I want to force players to make interesting choices.... like how they approach it, which rooms they tackle, at what point they withdraw, etc. I want those to be choices you have to think about each and every time you go into the depot. I don't want there to be any clear "best" strategy, especially not one you can use repeatedly every round. Make sense?

KingPhilipIII commented:
Yea I get it. Break up the meta, let every assault on the depot be fluid and requiring new strategies to cope with the random challenges since there is no best way to max out your loot.

Kyep commented:
Exactly. Now, if you have a suggestion for something that provides an interesting *choice* without being a clear "best" strategy, I'm totally open to that. For example, I'm still considering, at least a little, the relay v officer tradeoff idea. But there has to be a tradeoff involved, and the value of each option has to be relatively close, such that different players could go either way on the decision.

taukausanake commented:
I don't think there is an AI core in this map...

KingPhilipIII commented:
I’m about as creative as your average fresh water bass, so don’t expect many suggestions from me. 

Unrelated but i feel like the Syndie pod should be armed in some manner, since I assume in an IC sense they use it as a patrol craft for the depot.

Kyep commented:
@taukausanake It had been a couple of commits since I updated the screenshot. I have updated the screenshot again now. The AI core is in the SE of the map, next to the bathroom. It uses the fire graphic. This screenshot also shows the new loot system, with loot spawns being divided into levels 1-4, with 1 being worst and 4 being best. 

@KingPhilipIII Yeah... the issue with arming the space pod is that the pod is already quite strong, and easy to get to. Its one of the most accessible pieces of loot. I can't move it further in, because it requires room to move around. Aside from being one of the most accessible loot options, it is also quite strong. It has more health than a regular pod, and a snazzy paint scheme. Giving it weapons on top of both of those things might be.... too strong. How's a pod pilot meant to fight someone in a pod that has syndicate weapons plus more health? I worry they couldn't.  So, I am keeping the pod weaponless for now. I figure that it can serve as a stylish taxi to get someone back to the station. If people want to arm it, they can, but it will take extra effort. I don't want something that powerful to be easy to obtain.

On another note, currently, I am not doing anything special/unique with the syndie AI core that supposedly runs the depot. It is essentially just an NPC AI core that raises the alert if you destroy it. Now, I do have a bunch of structures which are destructible, and which raise the alert when destroyed. The purpose of these is to cause the alert level to spike up quickly if you start using bombs in the depot. But... I would like to do something special beyond this with the AI core. Any suggestions?

KingPhilipIII commented:
Maybe tie it into the lockdown that occurs when delta is reached? Also you could have Syndicate pod weapons spawn in the level four spawners so if people are feeling lucky they can make a go for it.

Kyep commented:
Tie it how?

KingPhilipIII commented:
Dunno. Was just thinking of it in the vein of the officer in the telecoms shed effect. 

Also, maybe want to make certain rooms unfriendly to thermite? Fast, efficient way to escape a lockdown, especially if you’ve cleared the rooms so you can just meth bolt your way out.

Kyep commented:
I am OK with people trying to use thermite to escape a lockdown. Door lockdowns only happen on delta. I don't think there is enough time for someone to thermite through every wall between one of the core rooms and the outside, before it blows. They're welcome to try, but I doubt it will work. To escape on delta, you really need to destroy walls in advance, and have a clear exit path already set up before delta is declared. If you are going to be destroying walls in advance, then thermite is certainly more fun to use than manually deconstructing each one. So, I think that is fine. 

Just bear in mind that you may not want to start deconstructing walls in the depot. The depot has no atmos system. It is surrounded by space. Even the sandy bit inside the asteroid has no air. Its mobs are also NOT immune to space. Result? Breaching the hull inside the depot may result in air draining out of all areas of it that are accessible from the bit you breached. Aside from issues with space wind, syndie mobs might suffocate. If they're officers, that raises the alert level. It is one of the counters I included to discourage people from simply using bombs to clear the depot.

KingPhilipIII commented:
Fair enough. This’ll definitely be an interesting spectacle for the first few weeks while watching people go at it.

KingPhilipIII commented:
Can cyborgs interact with the doors or no? Or does bringing them raise the alert?

Kyep commented:
@KingPhilipIII Silicons can override the locks on doors. They cannot override the locks on computers or other devices. Bringing them does not trigger anything special, as they are not considered massively more threatening than organic crew are.

KingPhilipIII commented:
Is the AI being named Rodger a reference to something?

Kyep commented:
@KingPhilipIII No. I was trying to think of a name for a Syndicate AI. I knew it would have to include the letters "S" (for syndicate) or "R" (for "red"), plus something about the depot ("D) plus maybe something about management or operations ("M" or "O").
Combine those and you get "R" + "O" + "D". "RODGER" was the natural choice. I could also have gone with "SMAD" or similar, but I thought "RODGER" was better.

KingPhilipIII commented:
If I can think of a more clever name would you change it?

Kyep commented:
No guarantees, but I'd think about it, yes. I would really like to do a bit more with the depot AI. Have the alert level system interact with it a bit more. Make it feel more like a mob and less like a passive mainframe. But, equally, that's getting into diminishing returns territory.

KingPhilipIII commented:
I mean. Let’s really consider what it would be doing in the event of a security breach. Probably tracking the attackers, attempting to impede their progress, and reporting data on them. Maybe have it’s destruction matter for that.

Kyep commented:
Its destruction already raises the alert level, similar to how NT raises the alert level on its stations when syndicate invaders take out its AI. 

I am not keen on the idea of destroying it preventing alert level rising, or otherwise hampering the syndicate, because that creates a powergame strategy whereby you can sneak in, take it out first, and then walk through the rest of the depot on easy mode, every time.

If I was going to have some benefit to destroying it, then I would need to randomize its location, make it harder to cheese, and possibly have a downside too so that it wasn't always the obvious route to take. In general, there should be no obvious routes to victory in the depot. It should always be a case of having to make decisions, tradeoffs, etc. There should never be a 'win button'.

KingPhilipIII commented:
I’m not saying destroying it should be a win button, but having something be distinctly beneficial so long as it’s hard enough to do that it justifies the benefit isn’t inherently a bad thing.

Kyep commented:
That's the issue. Right now, the AI core is always in the same room. It has no special defenses. Yeah, it has an officer in that room, BUT you have to kill at least one officer anyway. So IF I put in a benefit to killing the AI core... everyone who understands how the mission works will always take that room, and kill the core, because taking that room gets them armory access AND lets them access the benefit of killing the AI. Which is objectively a better option 100% of the time than trying to take one of the other officer rooms.

In order to add any sort of reward for killing the AI, I need to rethink the map layout... and have a dedicated AI core room which has its own defenses, and no reward other than killing the AI. That means that killing the AI will have an element of risk v reward. You have to risk taking on the defenses in that room, which you can normally ignore.... in exchange for the possible reward of whatever you get from killing the AI. That would be interesting.
But, supporting that requires me to create a new AI core room with its own (difficult!) defenses. And laying it out in such a way that its harder than most other rooms (because the reward from killing the AI in that scenario will be substantial). And replacing the AI core as a trap in that officer room.

In short, it isn't so easy. I *do* want to do it, I'm just not sure its practical. It is rather like the old idea of having a syndie comms relay you can destroy. Yeah, its doable. But balancing it is really hard, because if its good enough to be something you want to do, what stops you from doing it on every single depot run and just lowering the difficult/randomness of the depot in general?

KingPhilipIII commented:
I definitely get it maybe when after this gets merged when people spend the next few weeks smashing their heads against this because “New stuff” you’ll be able to get some feedback on it. I love the PR as is but some actual feedback would probably be the best thing to make good, balanced choices at this point.

Kyep commented:
Review please? This has been ready for merge for awhile.

Kyep commented:
Close/reopen to force travis update.

KasparoVy commented on code/modules/mob/living/simple_animal/hostile/syndicate.dm:
```diff
@@ -165,14 +167,15 @@
 	icon_living = "syndicatemeleespace"
 	maxHealth = 250
 	health = 250
+	melee_block_chance = 80
```
This man is ROBUST

Kyep replied:
It used to be they were all effectively immune to melee damage. Now, I've made it scale - the further into the depot you go, the harder the mobs get - in resistance to melee, and other ways.

Kyep commented:
Close/reopen to push map diff bot to get its act together.

Fox-McCloud commented on code/game/area/Space Station 13 areas.dm:
```diff
@@ -544,10 +544,333 @@ var/list/ghostteleportlocs = list()
+		announce_here("Depot Communications Offline", "Comms computer is damaged, destroyed or depowered. Unable to call in backup from Syndicate HQ.")
+	updateicon()
+
+/area/syndicate_depot/proc/activate_self_destruct(var/reason, var/containment_failure, var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/area/Space Station 13 areas.dm:
```diff
@@ -544,10 +544,333 @@ var/list/ghostteleportlocs = list()
+	for(var/obj/machinery/bluespace_beacon/syndicate/B in src)
+		return B.toggle()
+
+/area/syndicate_depot/proc/announce_here(var/a_header = "Depot Defense Alert", var/a_text = "")
```
*boop*

Fox-McCloud commented on code/game/area/Space Station 13 areas.dm:
```diff
@@ -544,10 +544,333 @@ var/list/ghostteleportlocs = list()
+		qdel(thismob)
+	guards_spawned = list()
+
+/area/syndicate_depot/proc/ghostlog(var/gmsg)
```
*boop*

Fox-McCloud commented on code/game/area/Space Station 13 areas.dm:
```diff
@@ -544,10 +544,333 @@ var/list/ghostteleportlocs = list()
+		run_finished = TRUE
+		log_game("Depot run: finished successfully: " + list_hostiles())
+
+/area/syndicate_depot/proc/add_hostile(var/mob/M)
```
*boop*

Fox-McCloud commented on code/game/machinery/Beacon.dm:
```diff
@@ -10,44 +10,62 @@
-	update_icon()
+
+/obj/machinery/bluespace_beacon/syndicate
+	syndicate = 1
```
`TRUE`

Fox-McCloud commented on code/game/machinery/doors/airlock_types.dm:
```diff
@@ -329,6 +329,20 @@
+	overlays_file = 'icons/obj/doors/airlocks/vault/overlays.dmi'
+	assemblytype = /obj/structure/door_assembly/door_assembly_vault
+	security_level = 6
+	hackProof = 1
```
`TRUE`

Fox-McCloud commented on code/game/machinery/doors/airlock_types.dm:
```diff
@@ -329,6 +329,20 @@
+	assemblytype = /obj/structure/door_assembly/door_assembly_vault
+	security_level = 6
+	hackProof = 1
+	aiControlDisabled = 1
```
`TRUE`

Fox-McCloud commented on code/game/machinery/navbeacon.dm:
```diff
@@ -213,3 +213,11 @@ Transponder Codes:<UL>"}
+
+
+/obj/machinery/navbeacon/invisible
+	invisibility = 101
```
This may be a define somewhere.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	name = "50pc int turret"
+	icon_state = "x"
+	result = list(
+	/obj/machinery/porta_turret/syndicate/interior  = 1
```
If you have a list with one thing in it, please don't new line it and put the ending `)` on another new line. It looks bad.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	/obj/item/clothing/under/syndicate = 1,
+	/obj/item/clothing/under/syndicate/tacticool = 1,
+	/obj/item/clothing/mask/gas/syndicate = 1,
+	/obj/item/storage/box/syndie_kit/cutouts = 1,
```
the comma at the last is unnecessary.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	/obj/item/clothing/under/syndicate/tacticool = 1,
+	/obj/item/clothing/mask/gas/syndicate = 1,
+	/obj/item/storage/box/syndie_kit/cutouts = 1,
+	)
```
put this after the `1` in the above line.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	icon_state = "x3"
+	// Loot schema: costumes, toys, useless gimmick items, trapped items
+	result = list(
+	/obj/item/storage/toolbox/syndicate = 1,
```
I'd prefer for lists like this, that the first item is on the same line as the list, then all others are indented to it, rather than it just be an open parenthesis then a new line for the next object. Nitpicky, but I think it looks cleaner.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	/obj/item/storage/box/syndie_kit/cutouts = 1,
+	)
+
+/obj/effect/spawner/random_spawners/syndicate/loot/spawn_item(var/turf/T)
```
*boop*

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	name = "rare loot"
+	// Loot schema: space gear, basic armor, basic ammo (10mm, rcd), drugs, more dangerous/useful gimmick items, lower-value minerals
+	result = list(
+	/obj/item/storage/box/syndie_kit/space = 1,
```
see above.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+/obj/effect/spawner/random_spawners/syndicate/loot/level3
+	name = "officer loot"
+	// Loot schema: medkits, very useful devices (jammer, illegal upgrade, RCD), better quality ammo (AP, fire), basic weapons (pistol, empgrenade), high value ores (diamond, uranium)
+	result = list(
```
here too.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+	name = "armory loot"
+	spawn_nothing_percentage = 0
+	// Loot schema: high-power weapons (m90, esword, ebow, revolver), devices that negate depot challenges (thermal glasses, chameleon device), explosives
+	result = list(
```
and here.

Fox-McCloud commented on code/game/objects/effects/spawners/random_spawners.dm:
```diff
@@ -108,3 +112,223 @@
+/obj/effect/spawner/random_spawners/syndicate/layout/door
+	name = "50pc door 25pc falsewall 25pc wall"
+	spawn_nothing_percentage = 0
+	result = list(
```
You get the idea for the rest.

Fox-McCloud commented on code/game/objects/items/weapons/teleportation.dm:
```diff
@@ -108,9 +108,11 @@ Frequency:
 	var/turf/current_location = get_turf(user)//What turf is the user on?
-	if(!current_location||!is_teleport_allowed(current_location.z))//If turf was not found or they're somewhere teleproof
+	var/area/current_area = get_area(user)
+	if(!current_location||!is_teleport_allowed(current_location.z) || current_area.tele_proof)//If turf was not found or they're somewhere teleproof
```
spacing here.

Fox-McCloud commented on code/game/objects/structures/depot.dm:
```diff
@@ -0,0 +1,100 @@
+	obj_integrity -= 10 * severity
+	healthcheck()
+
+/obj/structure/fusionreactor/bullet_act(var/obj/item/projectile/Proj)
```
*boop*

Fox-McCloud commented on code/game/objects/structures/depot.dm:
```diff
@@ -0,0 +1,100 @@
+			for(var/mob/living/M in range(10, loc))
+				M.apply_effect(rand(5, 25), IRRADIATE)
+	else
+		return ..(I, user, params)
```
`else return ..()`

Fox-McCloud commented on code/game/objects/structures/depot.dm:
```diff
@@ -0,0 +1,100 @@
+	else
+		return ..(I, user, params)
+
+/obj/structure/fusionreactor/proc/overload(var/containment_failure)
```
*boop*

Fox-McCloud commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -4,9 +4,53 @@
\ No newline at end of file
+	src.canmove = 0
+
+/mob/living/silicon/decoy/attackby(var/obj/item/W, var/mob/user, params)
```
*boop*

Fox-McCloud commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -4,9 +4,53 @@
+	if(istype(W, /obj/item/aicard))
+		user.visible_message("<span class='notice'>[user] cannot find an intellicard slot on [src].</span>")
+	else
+		return ..(W, user, params)
```
the vars on all `attackby` should be the same, almost without exception, this should be `return ..()`

Fox-McCloud commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -4,9 +4,53 @@
+	else
+		say("Connection failure!")
+
+/mob/living/silicon/decoy/syndicate/depot/death(var/pass)
```
*boop*

Fox-McCloud commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -4,9 +4,53 @@
+		raise_alert()
+	. = ..(pass)
+
+/mob/living/silicon/decoy/syndicate/depot/adjustBruteLoss(var/dmg)
```
*boop*

Fox-McCloud commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -4,9 +4,53 @@
+	. = ..(dmg)
+	updatehealth()
+
+/mob/living/silicon/decoy/syndicate/depot/adjustFireLoss(var/dmg)
```
*boop*

Fox-McCloud commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -4,9 +4,53 @@
+	. = ..(dmg)
+	updatehealth()
+
+/mob/living/silicon/decoy/syndicate/depot/ex_act(var/severity)
```
*boop*

Fox-McCloud commented on code/modules/mob/living/simple_animal/bot/medbot.dm:
```diff
@@ -90,6 +91,12 @@
 /mob/living/simple_animal/bot/medbot/syndicate/New()
 	..()
 	Radio.syndie = 1
+	faction |= "syndicate"
```
this should be on the medbot itself, not in `New`

Fox-McCloud commented on code/modules/mob/living/simple_animal/bot/syndicate.dm:
```diff
@@ -0,0 +1,215 @@
+			log_debug(failmsg)
+	return ..()
+
+/mob/living/simple_animal/bot/ed209/syndicate/proc/raise_alert(var/reason)
```
declare_crit

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+/obj/machinery/computer/syndicate_depot/emp_act(severity)
+	return
+
+/obj/machinery/computer/syndicate_depot/emag_act(var/mob/user)
```
declare_crit

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+
+/obj/machinery/computer/syndicate_depot/emag_act(var/mob/user)
+	to_chat(user, "<span class='notice'>The electronic systems in this console are far too advanced for your primitive hacking peripherals.</span>")
+	return
```
this isn't needed.

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		return 0
+	return ..()
+
+/obj/machinery/computer/syndicate_depot/proc/has_security_lockout(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	security_lockout = TRUE
+	disable_special_functions()
+
+/obj/machinery/computer/syndicate_depot/attack_hand(var/mob/user as mob)
```
*boop* no as mob either.

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	var/dat = get_menu(user)
+	user << browse(dat, "window=computer;size=575x450")
+	onclose(user, "computer")
+	return
```
stop this.

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		secondary(usr, text2num(href_list["secondary"]))
+	add_fingerprint(usr)
+	updateUsrDialog()
+	return
```
cease

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	return ..()
+
+
+/obj/machinery/computer/syndicate_depot/proc/get_menu(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+/obj/machinery/computer/syndicate_depot/proc/get_menu(var/mob/user)
+	return ""
+
+/obj/machinery/computer/syndicate_depot/proc/primary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		return 1
+	return 0
+
+/obj/machinery/computer/syndicate_depot/proc/secondary(var/mob/user, var/subcommand)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		return 1
+	return 0
+
+/obj/machinery/computer/syndicate_depot/proc/raise_alert(var/reason)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+/obj/machinery/computer/syndicate_depot/doors
+	name = "depot door control computer"
+
+/obj/machinery/computer/syndicate_depot/doors/get_menu(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	<BR><BR><a href='?src=[UID()];secondary=1'>Toggle Hidden Doors</a>
+	<BR>"}
+
+/obj/machinery/computer/syndicate_depot/doors/primary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		depotarea.toggle_door_locks(src)
+		to_chat(user, "<span class='notice'>Door locks toggled.</span>")
+
+/obj/machinery/computer/syndicate_depot/doors/secondary(var/mob/user, var/subcommand)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	req_access = list()
+	alerts_when_broken = TRUE
+
+/obj/machinery/computer/syndicate_depot/selfdestruct/get_menu(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		menutext += {"(ADMIN) Defense Shield: [depotarea.deployed_shields.len ? "ON" : "OFF"] (<a href='?src=[UID()];secondary=1'>[depotarea.deployed_shields.len ? "Disable" : "Enable"]</a>)<BR>"}
+	return menutext
+
+/obj/machinery/computer/syndicate_depot/selfdestruct/primary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		depotarea.activate_self_destruct("Fusion reactor containment field disengaged. All hands, evacuate. All hands, evacuate!", TRUE, user)
+
+
+/obj/machinery/computer/syndicate_depot/selfdestruct/secondary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	if(depotarea)
+		depotarea.comms_computer = src
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/get_menu(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		menu += "<BR><BR>Visiting Agents: NONE (<a href='?src=[UID()];secondary=[DEPOT_VISITOR_START]'>Sign-in as Agent</a>)"
+	return menu
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/primary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		raise_alert("Detected unauthorized access by [user] to [src]!")
+	updateUsrDialog()
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/secondary(var/mob/user, var/subcommand)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		to_chat(user, "<span class='warning'>ERROR: [src] is unable to uplink to depot network.</span>")
+	updateUsrDialog()
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/proc/grant_syndie_faction(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	name = "syndicate teleporter console"
+	icon_screen = "telesci"
+	icon_keyboard = "teleport_key"
+	var/obj/machinery/bluespace_beacon/syndicate/mybeacon
```
break the reference to this, if this is ever destroyed.

Kyep replied:
Added /obj/item/radio/beacon/syndicate/Destroy() in code/game/objects/items/devices/radio/beacon.dm that breaks this reference if destroyed.

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	var/obj/machinery/bluespace_beacon/syndicate/mybeacon
+	var/turf/tele_target
+	var/tele_area
+	var/obj/effect/portal/myportal
```
Same.

Kyep replied:
Is it enough to blank that when it is turned off (it turns off automatically when the computer is damaged or destroyed), or do you want me to separately make a subtype of portal that can hold a link back, and add a Destroy() proc for that subtype that calls back to clear the var?

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	icon_screen = "telesci"
+	icon_keyboard = "teleport_key"
+	var/obj/machinery/bluespace_beacon/syndicate/mybeacon
+	var/turf/tele_target
```
same.

Kyep replied:
What's the best way to do that? I can't really create a subtype of every possible turf with a backlink.

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		qdel(myportal)
+		myportal = null
+
+/obj/machinery/computer/syndicate_depot/teleporter/get_menu(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	to_chat(user, "<span class='notice'>Syndicate Teleporter Beacon: [bresult ? "<span class='green'>ON</span>" : "<span class='red'>OFF</span>"]</span>")
+	updateUsrDialog()
+
+/obj/machinery/computer/syndicate_depot/teleporter/secondary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+	icon_screen = "command"
+	req_access = list()
+
+/obj/machinery/computer/syndicate_depot/aiterminal/get_menu(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		menutext += "<BR><BR>Sentry Bot: (none present)"
+	return menutext
+
+/obj/machinery/computer/syndicate_depot/aiterminal/primary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/machinery/computer/depot.dm:
```diff
@@ -0,0 +1,481 @@
+		depotarea.reset_alert()
+		to_chat(user, "Alert level reset.")
+
+/obj/machinery/computer/syndicate_depot/aiterminal/secondary(var/mob/user)
```
*boop*

Fox-McCloud commented on code/game/area/Space Station 13 areas.dm:
```diff
@@ -544,10 +544,333 @@ var/list/ghostteleportlocs = list()
+	var/obj/structure/fusionreactor/reactor
+	var/list/hostiles_dead = list()
+	var/list/deployed_shields = list()
+	var/list/alert_log = list()
```
This area has a lot of held references; it's kinda ick because it means when these objects are deleted, they won't garbage collect, as they'll be held in this list. Please make sure references are cleared so things garbage collect.

Kyep replied:
/obj/structure/fusionreactor/Destroy() added.

I'm not sure how to handle hostiles_dead. On the one hand I do want it to be a list of /mobs, because its used to track who the mobs have seen die. On the other hand, I'm not sure how I can automatically remove /mobs from this list when they're deleted. Is there a known-good way to do that? or should I just have a timer that, once ever so often, deletes references to mobs that don't exist anymore?

deployed_shields automatically clears itself completely on proc/shields_down()

alert_log does not contain any references, it is simply a set of strings.

Crazylemon64 commented on code/game/atoms_movable.dm:
```diff
@@ -429,3 +429,7 @@
 	animate(I, alpha = 175, pixel_x = 0, pixel_y = 0, pixel_z = 0, time = 3)
+
+
+atom/movable/proc/portal_destroyed(obj/effect/portal/P)
```
Add a preceding `/` here
